### DISCUSSION TITLE
How can I use my custom pyrightconfig.json?

### DISCUSSION THREAD

**Discussion by yuxino** [TypeFox/monaco-languageclient#738] [Q&A]
I've been researching for a long time but haven't found a solution. I’ve tried placing the file in many different locations, but it still doesn’t work ... 😭

**Comment by CGNonofr** [comment]
Since there is no real context, I will assume you run your pyright language server on a server and connect to it via WebSocket

The pyrightconfig.json should probably be at the root directory of your project, on the server

**Comment by yuxino** [comment]
Thanks for your reply. I am trying to build an online editor.

Yes, my pyright language server is on the server and connected to it via WebSocket.

I referred to the [example](https://github.com/TypeFox/monaco-languageclient/tree/main/packages/examples/src/python) of monaco-languageclient and successfully built the project. It is a great project and saved me a lot of time.

I am a python newbie and don't know much about python, pyright or even lsp. But I want to achieve the following effect⬇️....

This editor can provide type hints but does not require explicit import of types. Like this 😂

<img width="357" alt="image" src="https://github.com/user-attachments/assets/9216d393-0a1d-4dd8-9b37-ec42c140569b">

Through the documentation of chatgpt and pright, I learned that `pyrightconfig.config` seems to be able to configure some built-in classes to achieve this effect. But I found that my lsp server could not read my configuration anyway 😭 So I asked this question ..

-----

Finally, I used a strange way to achieve this effect

``` tsx
<MonacoEditorReactComp
	// ...
	onLoad={(wrapper) => {
		setTimeout(() => {
			wrapper.editorApp.editor.setValue(`${builtins}\n${code}`);
			const hiddenRange = new monaco.Range(1, 1, 18, 1);
			wrapper.editorApp.editor.setHiddenAreas([hiddenRange])
		}, 0)
	}}
/>
```

Specifically, I hid the context. .

Because the editor has a complete context, it can prompt for input normally. .

![image](https://github.com/user-attachments/assets/bc9e915a-2b31-4f15-b650-40f591420166)

But I don't want to show this bunch of contextual tips to the user, so I hid the code, but hiding the code will cause some problems with the line number .. 

![image](https://github.com/user-attachments/assets/ab2fe261-4b32-4e31-b062-6b5075d02b22)

so I hid the line number in the UI 🤭 ....  

To sum up. . I want to implement such an editor, but I don't have any ideas. . .

Thank you again for reading such a long question 🙏

Can anyone tell me the correct way to do it? ... 🙏 🙏 🙏

**Reply by kaisalmen** [reply]
@yuxino does whatever you want to achieve work with the pyright extension in VSCode?

**Reply by yuxino** [reply]
![image](https://github.com/user-attachments/assets/6ebed18b-e41f-455c-97fb-fc4225b6ca6f)

hi. thanks for your reply, it seems that it is possible in vscode, using a file called __builtins__.pyi, the document reference is here https://microsoft.github.io/pyright/#/builtins.

Is there a way to achieve the same effect in monaco-languageclient? I think it is possible, but I don't know how to do it.

I thought maybe I could load builtins.pyi in the monaco-languageclient client? But I don't know how to load the file, nor do I know what API is available.

Almost all my knowledge about monaco, python, and lsp comes from the demos you wrote. Thank you again for your work.

**Reply by kaisalmen** [reply]
@yuxino you can check how the do it here:
https://github.com/microsoft/pyright/blob/main/packages/vscode-pyright/src/extension.ts

The example here show you how you can get started. We can't implement/maintain all the features of the existing language clients, so if you want more features you have to try yourself. If you want to contribute improvements here we are happy to review PRs. 🙂

**Reply by yuxino** [reply]
Thanks for your reply, I'll read the code and see if I can do it.
